### PR TITLE
Skip gcsfs registration to dask if dask is not installed

### PR DESCRIPTION
**Pull request by qibin-google**

### CODE REVIEW

File: gcsfs/dask_link.py [hunk 1]
```diff
@@ -118,7 +120,6 @@ def walk(self, path, refresh=False):
     dask.bytes.core._filesystems['gs'] = DaskGCSFileSystem
 
 
-try:
+# Register to dask if dask is installed
+if 'dask' in sys.modules:
```
Comment (martindurant) [inline gcsfs/dask_link.py]: This doesn't do what you think, it requires dask to have been already imported. Therefore, if gcsfs happens to be imported before dask, register will never be called.

File: gcsfs/dask_link.py [hunk 2]
```diff
@@ -118,7 +120,6 @@ def walk(self, path, refresh=False):
...
     register()
-except ImportError as e:
-    print(e)
```
Comment (martindurant) [inline gcsfs/dask_link.py:124]: I suggest the easiest thing to do here is to change the print to a logger.debug statement, so it doesn't normally show unless the user enables debug logging.

Comment (qibin-google) [reply gcsfs/dask_link.py:124]: Hi Martin,

Thanks for pointing it out. Yes, logger.debug will be much easier. Thanks for the suggestion.

Best,

Qibin

File: gcsfs/dask_link.py [hunk 3]
```diff
@@ -1,5 +1,10 @@
+import logging
+
 from .core import GCSFileSystem
 
+logging.basicConfig()
```
Comment (martindurant) [inline gcsfs/dask_link.py]: this should be up to the user - for gcsfs, it gets done if the environment variable is set on import

File: gcsfs/dask_link.py [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
+logger = logging.getLogger('dask_link')
```
Comment (martindurant) [inline gcsfs/dask_link.py]: "gcsfs" - this is then consistent throughout the package

File: gcsfs/dask_link.py [hunk 5]
```diff
@@ -121,4 +126,4 @@ def walk(self, path, refresh=False):
 try:
     register()
 except ImportError as e:
-    print(e)
+    logger.debug(e)
```
Comment (martindurant) [inline gcsfs/dask_link.py:128]: should be a string

File: gcsfs/dask_link.py (hunk 3 above)
Comment (qibin-google) [reply gcsfs/dask_link.py]: deleted

File: gcsfs/dask_link.py (hunk 4 above)
Comment (qibin-google) [reply gcsfs/dask_link.py]: Done.

File: gcsfs/dask_link.py (hunk 5 above)
Comment (qibin-google) [reply gcsfs/dask_link.py:128]: Here is an implicit conversion like the current implementation (print(e)). Maybe no need to add str(e). I can change it if you insist.

Comment (martindurant) [reply gcsfs/dask_link.py:128]: Seems you are correct, sorry for the noise.

**Pull request merged**